java/client/src/main/java/com/youtube/vitess/client/Proto.java

```diff
@@ -203,6 +210,26 @@ public EntityId apply(Map.Entry<byte[], ?> entry) {
         // Float, Double
         this.type = Query.Type.FLOAT64;
         this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Date) {
+        // Date
+        this.type = Query.Type.DATE;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Time) {
+        // Time
+        this.type = Query.Type.TIME;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Timestamp) {
+        // DateTime, TimeStamp
+        this.type = Query.Type.TIMESTAMP;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Boolean ) {
+        // Boolean
+        this.type = Query.Type.BIT;
+        this.value = ByteString.copyFromUtf8(((boolean)value) ? "1" : "0");
```

Comment:
boolean field is true or false so it has to be converted to 1 or 0.

As suggested I can use Query.Type.UINT64 for Boolean type but still conversion is required.